absorb database into common

iggi42 opened the pull request:
was to lazy to make a new branch, just gonna continue working on it. till the tag editor stands roughly.

iggi42 committed:
npm audit fix

iggi42 committed:
npm audit fix

iggi42 committed:
config tag_editor on port 4040 in dev

iggi42 committed:
config tag_editor on port 4040 in dev

iggi42 committed:
setup up the controller for modlists

iggi42 committed:
setup up the controller for modlists

iggi42 committed:
improve database setup

iggi42 committed:
improve database setup

iggi42 committed:
add modlists to the data model

iggi42 committed:
add modlists to the data model

iggi42 committed:
(try to) fix abc_seeds

iggi42 committed:
(try to) fix abc_seeds

iggi42 committed:
(try to) fix abc_seeds

iggi42 committed:
(try to) fix abc_seeds

iggi42 committed:
make ModFile a simple string (e.g. a url)

iggi42 committed:
make ModFile a simple string (e.g. a url)

iggi42 committed:
fix tags in abc_seeds (and merge local)

iggi42 committed:
remove unused variables in abc_seeds

iggi42 committed:
remove unused variables in abc_seeds

iggi42 committed:
Merge branch 'feature/basic-tag-editor' of github.com:Belethors-General-Mods/belethor into feature/basic-tag-editor

iggi42 committed:
remove unnecessary relation from mod_file to mod for embedding

iggi42 committed:
remove unnecessary relation from mod_file to mod for embedding

iggi42 committed:
remove useless comments

iggi42 committed:
remove useless comments

iggi42 committed:
add schema <-> struct transformation and make ModFile naming consistent.

iggi42 committed:
add schema <-> struct transformation and make ModFile naming consistent.

iggi42 committed:
make "desc" used consistent as "description"

iggi42 committed:
make "desc" used consistent as "description"

iggi42 committed:
correct function name

iggi42 committed:
correct function name

iggi42 committed:
start working on the database interface itself

iggi42 committed:
start working on the database interface itself

iggi42 committed:
Merge branch 'feature/basic-tag-editor' of github.com:Belethors-General-Mods/belethor into feature/basic-tag-editor

iggi42 committed:
Merge branch 'feature/basic-tag-editor' of github.com:Belethors-General-Mods/belethor into feature/basic-tag-editor

iggi42 committed:
absorb database into common

iggi42 committed:
update docs

iggi42 committed:
well, some docs were missing

iggi42 committed:
fix import warnings

iggi42 committed:
mix format

skwerlman commented:
travis is currently failing b/c of two things:

- dialyzer: unknown type Common.Struct.Mod.t/0
- mix ecto.create is failing because it cant talk to the database. this also causes all other db actions to fail, and causes tests and coverage to fail

iggi42 committed:
fix dialyzer warning

iggi42 committed:
update docs

iggi42 committed:
add postgres to travis

why ever the fuck it was not configured before

skwerlman commented:
travis is failing b/c coveralls is 500ing (no way to fix that on our end /shrug)
there are a couple warnings when running the tests about missing behaviours for the crawler
with the warnings fixed id be fine with merging without coveralls as long as coverage doesnt dip when checked manually (prev was 72%)

iggi42 committed:
add modlist fully to the database

iggi42 committed:
npm update

iggi42 committed:
add bgm logo from design

iggi42 committed:
working on mod crud

iggi42 committed:
working modlist

iggi42 committed:
add elm to tag editor

iggi42 committed:
wip switching workstation

iggi42 committed:
Merge branch 'feature/basic-tag-editor' of github.com:Belethors-General-Mods/belethor into feature/basic-tag-editor

iggi42 committed:
ignore elm-stuff

iggi42 committed:
more progress on the elm code base

iggi42 committed:
ignore elm-stuff folder

iggi42 committed:
more elm webform wip

iggi42 committed:
Merge branch 'feature/basic-tag-editor' of github.com:Belethors-General-Mods/belethor into feature/basic-tag-editor

skwerlman committed:
fix npm vulns, node 12 compat

iggi42 committed:
upgrade elm webpack loader to fix a vuln

iggi42 committed:
wip on elm form to edit a mod. (mostly done)

Bootstrap layout needs to be done.
Initial loading from json needs to be done.

iggi42 committed:
dirty push to mess with

iggi42 committed:
fix form with (a little dirty) jsonp trick.

iggi42 committed:
add real urls for my test abc seed.

iggi42 committed:
fix checkboxes and add attr path to the elm form

iggi42 committed:
fix checkboxes in form put

iggi42 committed:
enable debugging verbose logging in development env

iggi42 committed:
represent the boolean values as html selects

iggi42 committed:
use single select radio buttons for boolean attributes in form

iggi42 committed:
add a little documentation to Utils/Html.elm

iggi42 committed:
allow modfiles to be deleted

iggi42 committed:
mix format

iggi42 committed:
delete unrequired templates

iggi42 committed:
start using tags in the elm webform (not done)

iggi42 committed:
remove not required template

iggi42 committed:
make tagging mods functional (but butthurt ugly)

in other words: the tag can be added and removed to a mod, with this commit.
the ux is terrible thu .)

iggi42 committed:
upgrade to nodejs12 and install elm-format and elm-test

iggi42 committed:
add json encoder

iggi42 committed:
some elm test stuff (wip)

iggi42 committed:
fix stupid error

iggi42 committed:
json encode and decode fuzz tests

iggi42 committed:
make travis use elm-test (please?)

iggi42 committed:
elm-format everything :)

iggi42 committed:
mix format

iggi42 committed:
another travis test

iggi42 committed:
try to make elm-test and elm-format work in travis

iggi42 committed:
install elm for elm-test and elm-format

iggi42 committed:
make elm-format error messages a human text not json

iggi42 committed:
revert pretty error msg for elm-format

iggi42 committed:
fix elm-format test cases

iggi42 committed:
fix a bunch of compiler warnings

iggi42 committed:
more small stuff

iggi42 committed:
fix credoo issues and remove hello world test

remaining TODO s are to be made into issues, and then deleted

iggi42 committed:
use form_for/4 for dialyzer

iggi42 committed:
update docs

iggi42 committed:
basic moddb implementation

iggi42 committed:
wip on ModDB and tests

iggi42 committed:
update docs

iggi42 committed:
wip on moddb and clean_changes

iggi42 committed:
fix changeset sets

iggi42 committed:
mix docs

iggi42 committed:
fix mod crud test

iggi42 committed:
mix format

iggi42 committed:
start working on relation changes

like ModTag, ModList, etc

iggi42 committed:
npm update && npm audit fix

iggi42 committed:
fix postgres authentication in the dev env

iggi42 committed:
update docs

iggi42 committed:
write moddb tests with string keys and implement the change protcol more
